src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.Properties.cs

```diff
@@ -40,17 +35,23 @@ public double Maximum
 			set => SetValue(MaximumProperty, value);
 		}
 
-		public static DependencyProperty MaximumProperty { get ; } =
+		public static DependencyProperty MaximumProperty { get; } =
 			DependencyProperty.Register(nameof(Maximum), typeof(double), typeof(NumberBox), new FrameworkPropertyMetadata(double.MaxValue, (s, e) => (s as NumberBox)?.OnMaximumPropertyChanged(e)));
 
 		public double Value
 		{
 			get => (double)GetValue(ValueProperty);
-			set => SetValue(ValueProperty, value);
+			set
+			{
+				if (!Value.IsNaN() || !value.IsNaN())
```

Comment:
If `value` is NaN and `Value` is a number, this sets the value, is that expected?

Reply:
I don't know.. :(

https://github.com/microsoft/microsoft-ui-xaml/blob/09267531b807707138addc9bef3c8e9fa340615a/dev/NumberBox/NumberBox.cpp#L113

Reply:
Weird but it's MUX code so let's trust it 😂